torchtune/training/checkpointing/_checkpointer.py

```diff
@@ -146,31 +161,32 @@ def __init__(
         self._checkpoint_dir = Path(checkpoint_dir)
         self._checkpoint_path = get_path(self._checkpoint_dir, checkpoint_files[0])
 
-        if not self._checkpoint_path.suffix == ".pt":
-            raise ValueError(
-                f"Checkpoint file {self._checkpoint_path} is not a valid checkpoint file. "
-                "Checkpointer expects a valid .pt file."
-            )
-
-        self._adapter_checkpoint = (
-            get_path(self._checkpoint_dir, adapter_checkpoint)
-            if adapter_checkpoint
-            else None
-        )
-
         self._resume_from_checkpoint = resume_from_checkpoint
         self._model_type = ModelType[model_type]
         self._output_dir = Path(output_dir)
+        self._output_dir.mkdir(exist_ok=True)
 
-        # recipe_checkpoint contains the recipe state. This should be available if
-        # resume_from_checkpoint is True
-        self._recipe_checkpoint = None
-        if self._resume_from_checkpoint:
-            if recipe_checkpoint is None:
-                raise ValueError(
-                    "If resume_from_checkpoint is True, recipe_checkpoint file must be provided."
-                )
-            self._recipe_checkpoint = get_path(self._checkpoint_dir, recipe_checkpoint)
+        # save all files in input_dir, except model weights and mapping, to output_dir
+        # this is useful to preserve the tokenizer, configs, license, etc.
+        copy_files(
+            self._checkpoint_dir,
+            Path.joinpath(self._output_dir, BASE_MODEL_DIRNAME),
+            ignore_suffixes=SUFFIXES_TO_NOT_COPY,
```

Comment:
Do we think that a single global variable for `SUFFIXES_TO_NOT_COPY` will suffice here? Maybe I'm paranoid but I anticipate this will be something people will need to hack into and change

Reply:
i should add to this function to not copy anything > x MiB, e.g. 200MiB. But i dont think that this should require user input. They have easily cp/rm files. Do you see an easy fix that doesnt greatly increase complexity?

Reply:
Yeah I agree with not increasing complexity here, I think it's OK to leave it like this for now